examples/CustomEntitlementComputationSample/app/src/main/java/com/revenuecat/sample/main/MainViewModel.kt

```diff
@@ -0,0 +1,115 @@
+package com.revenuecat.sample.main
+
+import android.app.Activity
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.PurchaseParams
+import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.PurchasesException
+import com.revenuecat.purchases.awaitOfferings
+import com.revenuecat.purchases.interfaces.UpdatedCustomerInfoListener
+import com.revenuecat.purchases.purchaseWith
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
+class MainViewModel : ViewModel() {
+
+    private val _uiState: MutableStateFlow<MainState> = MutableStateFlow(
+        MainState(),
+    )
+    val uiState: StateFlow<MainState> = _uiState.asStateFlow()
+
+    init {
+        Purchases.sharedInstance.updatedCustomerInfoListener =
+            UpdatedCustomerInfoListener { customerInfo ->
+                updateCustomerInfoInformation(customerInfo)
+            }
+        viewModelScope.launch {
+            getOfferings()
+        }
+    }
+
+    private suspend fun getOfferings() {
+        try {
+            val offerings = Purchases.sharedInstance.awaitOfferings()
+            _uiState.update { currentState ->
+                currentState.copy(
+                    offerings = offerings,
+                )
+            }
+        } catch (error: PurchasesException) {
+            _uiState.update { currentState ->
+                currentState.copy(
+                    displayErrorMessage = error.message,
+                )
+            }
+        }
+    }
+
+    fun switchUser(newUserID: String) {
+        Purchases.sharedInstance.switchUser(newUserID)
+    }
+
+    fun resetErrorMessage() {
+        _uiState.update { currentState ->
+            currentState.copy(
+                displayErrorMessage = null,
+            )
+        }
+    }
+
+    fun initiateSwitchUserProcess() {
+        _uiState.update { it.copy(shouldStartSwitchingUser = true) }
+    }
+
+    fun resetSwitchUserProcess() {
+        _uiState.update { it.copy(shouldStartSwitchingUser = false) }
```

Comment:
feels like we could follow this same one-liner format for the others without losing clarity?